tkg/client/update_cluster_metadata.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	crtclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/tanzu-framework/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/tkg/constants"
+)
+
+const clusterMetadataNamespace = "tkg-system-public"
+const tkgBomConfigMapName = "tkg-bom"
+const tkgMetadataConfigMapName = "tkg-metadata"
+const clusterMetadataNamespaceRoleName = "tkg-metadata-reader"
+
+// CreateOrUpdateClusterMetadata will create or update the tkg-bom ConfigMap and tkg-metadata ConfigMap
+func (c *TkgClient) CreateOrUpdateClusterMetadata(regionalClusterClient, currentClusterClient clusterclient.Client, isRegionalCluster bool, edition string, clusterName string, clusterNamespace string) error {
```

Comment:
Why can't metadata be a sub package of TKG meta package?

Reply:
Because everything in the meta package exists in the mgmt cluster, but cluster metadata is also in the workload cluster.

Reply:
It 's difficult to understand why it is required in each workload cluster. But my point is why can not bundle them as a Carvel package which is easy to manage?